karton/core/backend.py

```diff
@@ -273,6 +284,8 @@ def set_task_status(
         :param status: New task status (TaskState)
         :param pipe: Optional pipeline object if operation is a part of pipeline
         """
+        if task.status == status:
```

Comment:
Does this actually happen? Maybe we should introduce a small log.warning here to get rid of the redundant calls over time?

Reply:
It does happen, quite a lot actually (IIRC it reduced calls like 20%)